src/futures_map.rs

```diff
@@ -122,6 +122,39 @@ where
             }
         }
     }
+
+    /// Returns an iterator over all futures of type `T` pushed via [`FuturesMap::try_push`].
+    /// The order that futures are returned is not guaranteed.
+    ///
+    /// If downcasting a future to `T` fails it will be skipped in the iterator.
+    pub fn iter_of_type<T>(&self) -> impl Iterator<Item = (&ID, &T)>
+    where
+        T: 'static,
+    {
+        self.inner.iter().filter_map(|a| {
+            let pin = a.inner.inner.as_ref();
+            let any = Pin::into_inner(pin) as &(dyn Any + Send);
+            let inner = any.downcast_ref::<T>()?;
+            Some((&a.tag, inner))
+        })
+    }
+
+    /// Returns an iterator with mutable access over all futures of type `T` pushed via
+    /// [`FuturesMap::try_push`].
+    /// The order that futures are returned is not guaranteed.
+    ///
+    /// If downcasting a future to `T` fails it will be skipped in the iterator.
+    pub fn iter_mut_of_type<T>(&mut self) -> impl Iterator<Item = (&mut ID, &mut T)>
```

Comment:
I might have missed this in the other PR actually but I don't think we should be returning mutable references to the ID.

Reply:
Yeah, that did seem a bit weird. Rust's standard library types only allow the value to be mutated, except in special cases with a bunch of warnings.